code/clue.js

```diff
@@ -10,50 +10,205 @@ const mrGreen = {
+ 'Billiard room',
+ 'Library',
+ 'Lounge',
+ 'Hall',
+ 'Spa',
+]
+  
 
-const rooms = []
+// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION. (randomSelector)
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
 
-// This function will be invoked when you click on the killer card.
+const mystery = {
+  killer: "",
+  weapon: "",
+  room: ""
+}
+
+// This function will be invoked when clicking the killer card.
+
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  document.getElementById('killerImage').src = `${mystery.killer.image}`
   document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerName').innerHTML = (`Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`)
+  document.getElementById('killerAge').innerHTML = (`Age: ${mystery.killer.age}`)
+  document.getElementById('killerOccupation').innerHTML = (`Occupation: ${mystery.killer.occupation}`)
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+// This function will be invoked when clicking the weapon card.
+
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+  document.getElementById('weaponCard').style.background = mystery.weapon.color
+  document.getElementById('weaponName').innerHTML = mystery.weapon.name
+  document.getElementById('weaponWeight').innerHTML = `It weighs ${mystery.weapon.weight} kg.`
+}
+
+// This function will be invoked when clicking the room card.
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+  document.getElementById('roomName').innerHTML = mystery.room
+}
+
+document.getElementById('killerCard').onclick = pickKiller
+document.getElementById('weaponCard').onclick = pickWeapon
+document.getElementById('roomCard').onclick = pickRoom
+
 
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+const revealMystery = () => {
+  theMystery.innerHTML = (`The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName} with the ${mystery.weapon.name} in the ${mystery.room}!`)
```

Comment:
If you click reveal before clicking the cards, the message is pretty hard for the user to understand.  Is there a better way to do this?